File: 71-simplify-path/simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        int i = path.length()-1;
        int count = 0;
        string ans = "";
        while(i >= 0){
            string s = "";
            while(path[i] != '/'){
                s += path[i];
                i--;
            }
            if(s == "" || s == "."){
                i--;
                continue;
            }
            else if(s == "..") count++;
            else{
                if(count > 0) count--;
                else {reverse(s.begin(),s.end()); ans = "/" + s + ans;}
            }
            i--;
        }
        return ans == "" ? "/" : ans;
    }
};

File: 71-simplify-path/simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:

        # split the path with '/'
        arr = path.split('/')

        # create a stack and filter the results with empty content
        stack = []
        for ele in arr:
            if ele == "" or ele == ".":
                continue
            elif ele == "..":
                if len(stack) > 0:
                    stack.pop()
            else:
                stack.append(ele)
        
        final_path = '/'.join(stack)

        return '/' + final_path